packages/promotion/src/repositories/campaign.ts

```diff
@@ -8,4 +10,137 @@ export class CampaignRepository extends DALUtils.mikroOrmBaseRepositoryFactory<
     create: CreateCampaignDTO
     update: UpdateCampaignDTO
   }
->(Campaign) {}
+>(Campaign) {
+  async create(
+    data: CreateCampaignDTO[],
+    context: Context = {}
+  ): Promise<Campaign[]> {
+    const manager = this.getActiveManager<SqlEntityManager>(context)
+    const promotionIds: string[] = []
+    const campaignIdentifierPromotionsMap = new Map<string, string[]>()
+
+    data.forEach((campaignData) => {
+      const campaignPromotionIds =
+        campaignData.promotions?.map((p) => p.id) || []
+
+      promotionIds.push(...campaignPromotionIds)
+
+      campaignIdentifierPromotionsMap.set(
+        campaignData.campaign_identifier,
+        promotionIds
```

Comment:
question: shouldn't this be campaignPromotionIds?

Reply:
oh good catch!